src/golang/lib/collections/artifact/types.go

```diff
@@ -1,159 +1,17 @@
 package artifact
 
-import (
-	"database/sql/driver"
-	"encoding/json"
-
-	"github.com/aqueducthq/aqueduct/lib/collections/artifact/boolean"
-	"github.com/aqueducthq/aqueduct/lib/collections/artifact/float"
-	"github.com/aqueducthq/aqueduct/lib/collections/artifact/jsonable"
-	"github.com/aqueducthq/aqueduct/lib/collections/artifact/table"
-	"github.com/aqueducthq/aqueduct/lib/collections/utils"
-	"github.com/dropbox/godropbox/errors"
-)
-
-// This file covers all artifact specs.
-//
-// To add a new spec:
-// - Add a new enum constant for `Type`
-// - Add a new field in specUnion for the new spec struct
-// - Implement 3 additional methods for top level Spec type:
-//  - IsNewType() method to validate the type
-//  - NewType() method to get the value of the type from private `spec` field
-//  - NewSpecFromNewType() method to construct a spec from the new type
-
 type Type string
 
 const (
-	TableType Type = "table"
-	FloatType Type = "float"
-	BoolType  Type = "boolean"
-	JsonType  Type = "json"
+	Untyped       Type = "untyped"
+	StringType    Type = "string"
+	BoolType      Type = "bool"
+	NumericType   Type = "numeric"
+	DictType      Type = "dictionary"
+	TupleType     Type = "tuple"
+	TableType     Type = "table"
+	JsonType      Type = "json"
+	BytesType     Type = "bytes"
+	ImageType     Type = "image"
+	PicklableType Type = "picklable"
```

Comment:
nit: Do we need the "Type" suffix? I wonder if its necessary, for example:
```
func example(artifactType artifact.Type) {
    if artifactType == artifact.String {
          ....
    }
}
```
looks very reasonable to me.